builder/osc/chroot/step_create_omi.go

```diff
@@ -19,85 +18,85 @@ type StepCreateOMI struct {
 
 func (s *StepCreateOMI) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
 	config := state.Get("config").(*Config)
-	osconn := state.Get("osc").(*osc.APIClient)
+	osconn := state.Get("osc").(*(osccommon.OscClient))
 	snapshotId := state.Get("snapshot_id").(string)
 	ui := state.Get("ui").(packersdk.Ui)
 
+	//regionconn := config.NewOSCClientByRegion(region)
 	ui.Say("Creating the OMI...")
 
 	var (
-		registerOpts   osc.CreateImageRequest
-		mappings       []osc.BlockDeviceMappingImage
-		image          osc.Image
+		registerOpts   oscgo.CreateImageRequest
+		mappings       []oscgo.BlockDeviceMappingImage
+		image          oscgo.Image
 		rootDeviceName string
 	)
 
 	if config.FromScratch {
 		mappings = config.OMIBlockDevices.BuildOscOMIDevices()
 		rootDeviceName = config.RootDeviceName
 	} else {
-		image = state.Get("source_image").(osc.Image)
-		mappings = image.BlockDeviceMappings
-		rootDeviceName = image.RootDeviceName
+		image = state.Get("source_image").(oscgo.Image)
+		mappings = *image.BlockDeviceMappings
+		rootDeviceName = *image.RootDeviceName
 	}
 
-	newMappings := make([]osc.BlockDeviceMappingImage, len(mappings))
+	newMappings := make([]oscgo.BlockDeviceMappingImage, len(mappings))
 	for i, device := range mappings {
 		newDevice := device
 
 		//FIX: Temporary fix
-		gibSize := newDevice.Bsu.VolumeSize / (1024 * 1024 * 1024)
-		newDevice.Bsu.VolumeSize = gibSize
+		gibSize := *newDevice.Bsu.VolumeSize / (1024 * 1024 * 1024)
+		newDevice.Bsu.VolumeSize = &gibSize
 
-		if newDevice.DeviceName == rootDeviceName {
-			if newDevice.Bsu != (osc.BsuToCreate{}) {
-				newDevice.Bsu.SnapshotId = snapshotId
+		if newDevice.GetDeviceName() == rootDeviceName {
+			if *newDevice.Bsu != (oscgo.BsuToCreate{}) {
```

Comment:
If think, you just need to check that the newDevice.Bsu is nil or not. The v1 sdk, we were forced to check values, but now just check the pointer